components/esp_driver_i2c/i2c_master.c

```diff
@@ -784,40 +784,48 @@ static esp_err_t i2c_master_bus_destroy(i2c_master_bus_handle_t bus_handle)
 {
     ESP_RETURN_ON_FALSE(bus_handle, ESP_ERR_INVALID_ARG, TAG, "no memory for i2c master bus");
     i2c_master_bus_handle_t i2c_master = bus_handle;
-    i2c_common_deinit_pins(i2c_master->base);
-    if (i2c_release_bus_handle(i2c_master->base) == ESP_OK) {
-        if (i2c_master) {
-            if (i2c_master->bus_lock_mux) {
-                vSemaphoreDeleteWithCaps(i2c_master->bus_lock_mux);
-                i2c_master->bus_lock_mux = NULL;
-            }
-            if (i2c_master->cmd_semphr) {
-                vSemaphoreDeleteWithCaps(i2c_master->cmd_semphr);
-                i2c_master->cmd_semphr = NULL;
-            }
-            if (i2c_master->event_queue) {
-                vQueueDeleteWithCaps(i2c_master->event_queue);
-            }
-            if (i2c_master->queues_storage) {
-                free(i2c_master->queues_storage);
-            }
-            free(i2c_master->i2c_async_ops);
-            for (int i = 0; i < I2C_TRANS_QUEUE_MAX; i++) {
-                if (i2c_master->trans_queues[i]) {
-                    vQueueDelete(i2c_master->trans_queues[i]);
+
+    // Check if i2c_master->base is not NULL before using it
+    if (i2c_master->base) {
+        i2c_common_deinit_pins(i2c_master->base);
+
+        if (i2c_release_bus_handle(i2c_master->base) == ESP_OK) {
+            if (i2c_master) {
```

Comment:
this specific check for non-null is redundant at this point since the pointer will have been dereferenced already and has been validated already as part of the first line in the function.

Reply:
Good catch. Please see if latest looks correct.

Reply:
Looks much better!